diff --git a/rosen/modules/render_service/core/rs_render_pipeline_agent.cpp b/rosen/modules/render_service/core/rs_render_pipeline_agent.cpp
index d0fb4ca989..d45ae27269 100644
--- a/rosen/modules/render_service/core/rs_render_pipeline_agent.cpp
+++ b/rosen/modules/render_service/core/rs_render_pipeline_agent.cpp
@@ -1444,7 +1444,7 @@ void RSRenderPipelineAgent::CollectSurfaceBuffersByProcessId(
             auto surfaceBufferTuple = std::make_tuple(bufferInfo.surfaceBuffer_, surfaceName, absRect);
             sfBufferInfoVector.push_back(surfaceBufferTuple);
 
-            RS_LOGD("CollectSurfaceBuffersByProcessId: Added texture buffer from pid=%{public}d, uid=%{public}llu, "
+            RS_LOGD("CollectSurfaceBuffersByProcessId: Added texture buffer from pid=%{public}d, uid=%{public}" PRIu64 ", "
                     "bufferId=%{public}u, size=%{public}dx%{public}d",
                     bufferInfo.pid_, bufferInfo.uid_,
                     bufferInfo.surfaceBuffer_->GetSeqNum(),
diff --git a/rosen/modules/render_service_base/src/feature/buffer_reclaim/rs_buffer_reclaim.cpp b/rosen/modules/render_service_base/src/feature/buffer_reclaim/rs_buffer_reclaim.cpp
index 711ac26e68..a97259aa32 100644
--- a/rosen/modules/render_service_base/src/feature/buffer_reclaim/rs_buffer_reclaim.cpp
+++ b/rosen/modules/render_service_base/src/feature/buffer_reclaim/rs_buffer_reclaim.cpp
@@ -45,7 +45,7 @@ bool RSBufferReclaim::DoBufferReclaim(sptr<SurfaceBuffer> buffer)
     }
     bufferReclaimNumsSet_.insert(buffer->GetBufferId());
     RS_TRACE_NAME_FMT("DoBufferReclaim: bufferReclaimNumsSet_=%lu", bufferReclaimNumsSet_.size());
-    RS_LOGI("DoBufferReclaim: bufferReclaimNumsSet_=%{public}u", bufferReclaimNumsSet_.size());
+    RS_LOGI("DoBufferReclaim: bufferReclaimNumsSet_=%{public}zu", bufferReclaimNumsSet_.size());
 
     bool ret = false;
     if (buffer->TryReclaim() == GSERROR_OK) {
@@ -67,7 +67,7 @@ bool RSBufferReclaim::DoBufferResume(sptr<SurfaceBuffer> buffer)
     }
     bufferReclaimNumsSet_.erase(iter);
     RS_TRACE_NAME_FMT("DoBufferResume: bufferReclaimNumsSet_=%lu", bufferReclaimNumsSet_.size());
-    RS_LOGI("DoBufferResume: bufferReclaimNumsSet_=%{public}u", bufferReclaimNumsSet_.size());
+    RS_LOGI("DoBufferResume: bufferReclaimNumsSet_=%{public}zu", bufferReclaimNumsSet_.size());
     bool ret = false;
     if (buffer->TryResumeIfNeeded() == GSERROR_OK) {
         buffer->UnRegisterBufferDestructorCallBack();
@@ -82,7 +82,7 @@ bool RSBufferReclaim::CheckBufferReclaim()
 {
     std::lock_guard<std::mutex> lock(mutex_);
     RS_TRACE_NAME_FMT("CheckBufferReclaim: bufferReclaimNumsSet_=%lu", bufferReclaimNumsSet_.size());
-    RS_LOGI("CheckBufferReclaim: bufferReclaimNumsSet_=%{public}u", bufferReclaimNumsSet_.size());
+    RS_LOGI("CheckBufferReclaim: bufferReclaimNumsSet_=%{public}zu", bufferReclaimNumsSet_.size());
     return bufferReclaimNumsSet_.size() < MAX_BUFFER_RECLAIM_NUMS;
 }
 
@@ -92,7 +92,7 @@ void RSBufferReclaim::RemoveBufferReclaim(uint64_t bufferId)
     std::lock_guard<std::mutex> lock(mutex_);
     bufferReclaimNumsSet_.erase(bufferId);
     RS_TRACE_NAME_FMT("RemoveBufferReclaim: bufferReclaimNumsSet_=%lu", bufferReclaimNumsSet_.size());
-    RS_LOGI("RemoveBufferReclaim: bufferReclaimNumsSet_=%{public}u", bufferReclaimNumsSet_.size());
+    RS_LOGI("RemoveBufferReclaim: bufferReclaimNumsSet_=%{public}zu", bufferReclaimNumsSet_.size());
 }
 } // namespace Rosen
 } // namespace OHOS
diff --git a/rosen/modules/render_service_base/src/pipeline/rs_surface_buffer_callback_manager.cpp b/rosen/modules/render_service_base/src/pipeline/rs_surface_buffer_callback_manager.cpp
index 7080607fe1..8919564c37 100644
--- a/rosen/modules/render_service_base/src/pipeline/rs_surface_buffer_callback_manager.cpp
+++ b/rosen/modules/render_service_base/src/pipeline/rs_surface_buffer_callback_manager.cpp
@@ -434,7 +434,7 @@ void RSSurfaceBufferCallbackManager::RemoveAllSurfaceBufferInfo(pid_t pid, uint6
     auto it = surfaceBufferInfoMap_.find(key);
     if (it != surfaceBufferInfoMap_.end()) {
         RS_LOGD("RSSurfaceBufferCallbackManager::RemoveAllSurfaceBufferInfo:"
-                "Removed %{public}zu buffers for pid=%{public}d, uid=%{public}llu",
+                "Removed %{public}zu buffers for pid=%{public}d, uid=%{public}" PRIu64,
                 it->second.size(), pid, uid);
         surfaceBufferInfoMap_.erase(it);
     }
